dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/DSpaceConverter.java

```diff
@@ -15,7 +16,11 @@ public R convert(M source) {
 		return fromModel(source);
 	}
 
-	public abstract R fromModel(M obj);
+	public R fromModel(M obj) {
+		return fromModel(obj, DSpaceObjectRest.PRJ_DEFAULT);
```

Comment:
Our converters already have dependencies from other converters, see the ItemConverter for instance
https://github.com/DSpace/DSpace/blob/master/dspace-spring-rest/src/main/java/org/dspace/app/rest/converter/ItemConverter.java#L32
so I assume that it is possible switching to ConverterFactory instanciate a bitstream converter using a different projection than the one applied to our primary object. So I don't see any issue in implementing the projection scenario that you have described also using the option 1.

About option 2: yes it is possible to know if a field in an hibernate object is initialized or not without forcing lazy-loading, see for instance these stackoverflow questions
http://stackoverflow.com/questions/15890559/java-hibernate-how-to-detect-if-field-is-lazy-loaded-proxy-and-not-actual-data
http://stackoverflow.com/questions/21476600/hibernate-isinitialized-for-lazy-loaded-fields

I have to check which API we need to use for our exact hibernate version